tests/unit/torch/test_block.py

```diff
@@ -377,3 +377,58 @@ def forward(self, inputs: Dict[str, torch.Tensor]) -> Tuple[torch.Tensor, torch.
         shortcut_propagate = ShortcutBlock(PlusOneShortcutTuple(), propagate_shortcut=True)
         with pytest.raises(RuntimeError):
             module_utils.module_test(shortcut_propagate, torch.rand(5, 5))
+
+
+class TestBatchBlock:
+    def test_forward_with_batch(self):
+        batch = Batch(torch.tensor([1, 2]), torch.tensor([3, 4]))
+        outputs = mm.BatchBlock()(batch)
+
+        assert batch == outputs
+
+    def test_forward_with_features(self):
+        feat = torch.tensor([1, 2])
+        outputs = module_utils.module_test(mm.BatchBlock(), feat)
+        assert isinstance(outputs, mm.Batch)
+        assert torch.equal(outputs.feature(), feat)
+
+    def test_forward_with_tuple(self):
+        feat, target = torch.tensor([1, 2]), torch.tensor([3, 4])
+        outputs = module_utils.module_test(mm.BatchBlock(), feat, targets=target)
+
+        assert isinstance(outputs, mm.Batch)
+        assert torch.equal(outputs.feature(), feat)
+        assert torch.equal(outputs.target(), target)
+
+    def test_forward_exception(self):
+        with pytest.raises(RuntimeError):
```

Comment:
I noticed that in some places we use the  `match=` argument of `pytest.raises` to specify which error we're expecting. I think this is a good pattern to adopt and will make things more robust. Since this could start failing from a different place accidentally and still pass the test, changing what the test is checking.